### ISSUE TITLE
CLI JSON output

### ISSUE THREAD

**Issue by lkingland** [boson-project/faas#37]
Some of the commands support specifying JSON output; it would be nice to have all commands support this flag.

**Comment by lance** [comment]
I see that `describe` has YAML/JSON/XML output options, but it seems that may be the only command that does. How do you see this working for other commands? 

*  `create` optionally spits out YAML for the kn `Service`
* `delete` not sure about this one
* `list` YAML of the deployed `Service`(s)
* `run` ??
* `update` spits out YAML for the updated `Service`

Am I missing anything?

**Comment by lkingland** [comment]
As for the other commands, I was hoping we could create a flag which is short-hand for saying "human readable or machine readable".  In the case of `describe`, the YAML and XML options of course make sense.  Perhaps for the others only JSON does (essentially requesting structured logging)

`run` is a little bit of a special case, as it delegates...   We could respect this flag in faas itself, perhaps setting a well-known environment variable which can optionally be respected by the Function author to choose the way they treat STDOUT and STDERR?

**Comment by matejvasek** [comment]
@lkingland `list` currently returns only names. If we wanted return more, what fields are we interested in?

**Comment by lkingland** [comment]
It's my opinion that we should be seeking the simplest possible functional case at this time, and only add more as specific use-cases require them.  The minimum necessary information to subscribe to an event is its unique identifier, thus the minimum necessary information back out of the system would be those same unique identifiers returned.   If we want to add more information; perhaps augmented with fields that are platform-specific, I would see that as a GET request for more information about the subscription of that ID.
so POSTing a name to `localhost:1111/subscriptions` would append the name to the list.  The list would then be returned by GET, with additional information below the named event.  Something like:
```
GET localhost:1111/subscriptions
[ "newsletter-signups" ]
```
```
GET localhost:1111/subscriptions/newsletter-signups
{ "name": "newsletter-signups",
.... // Other fields lazily added here _only as required_ to fulfill specifically requested use-cases
}
```
By keeping ourselves so simple as to be nearly an oversimplification, I am hoping we can prevent from painting ourselves into a corner unnecessarily.
I may indeed be missing something here, so please let me know if there are other fields than the event's name which are required to implement the basic pub/sub flow!

**Comment by matejvasek** [comment]
@lkingland  by `list` here I meant list of deployed functions/ksvcs not subscriptions.

**Comment by lkingland** [comment]
OK, I see that's for the listing of deployed Functions.  I'd say the same general suggestion applies.  Just return a list of the names, and then one can get more information about a given service with `GET` for that specific service by name `/services/www.example.com`

**Comment by lkingland** [comment]
This is complete.  Implemented by @matejvasek in #67 and slightly tweaked to have format functions consolidated with the `describe` command in #79 .  Still outstanding is the issue of having structured logging such that _all_ output is of the explicit type, but if we need that, we can save that for a future discussion/info.
